infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironmentFactoryTest.java

```diff
@@ -297,6 +297,21 @@ public void exceptionOnRecipeLoadError() throws Exception {
     k8sEnvFactory.doCreate(internalRecipe, emptyMap(), emptyList());
   }
 
+  @Test(
+      expectedExceptions = ValidationException.class,
+      expectedExceptionsMessageRegExp =
+          "Environment can not contain two 'Service' objects with the same name 'db'")
```

Comment:
Minor suggestion: since we're using the regexp version, leave formatted parts in message ambiguous (`"Environment can not contain two .* objects with the same name .*"`)

Reply:
Personally, I prefer to leave a message with concrete values, since we use `String.format` for construction of messages and such check can help to catch out wrong parameters order.

Reply:
:+1: